getting-started/installation/general-installation/installation-community-containerized.markdown

```diff
@@ -0,0 +1,249 @@
+Docker contaiers will be created, one container to be the Policy Server (server), and another container that will be the Host Agent (client).
+
+Both will will run ubi9 images and communicate on a container network.
+Upon completion, you are ready to start working with CFEngine.
+
+## Requirements
+* 1G+ disk space
+* 1G+ memory
+* Working [Docker Engine](https://docs.docker.com/engine/) or [Podman](https://podman.io/) setups on a supported [x86_64](https://en.wikipedia.org/wiki/X86-64) platform.
+
+**Note**: This document considers [Docker Engine](https://docs.docker.com/engine/) for all examples.
+Use of [Podman](https://podman.io/) shall be similar with adequate adaptations. (_Ref_: [Emulating Docker CLI with Podman](https://podman-desktop.io/docs/migrating-from-docker/emulating-docker-cli-with-podman)).
+
+## Overview
+1. Installing container engine
+2. Preparing CFEngine hub in container
+3. Preparing CFEngine host in container
+4. Using docker compose
+    1. Preparing container image for CFEngine
+    2. Using docker compose service
+5. References
+
+## Installing container engine
+**Ref**: [Install Docker Engine](https://docs.docker.com/engine/install/)
+
+OR
+
+**Ref**: [Podman Installation Instructions](https://podman.io/docs/installation)
+(_Optionally_: [Emulating Docker CLI with Podman](https://podman-desktop.io/docs/migrating-from-docker/emulating-docker-cli-with-podman))
+
+## Preparing CFEngine hub in container
+Run the container with systemd
+
+```command
+$ docker run --privileged -dit --name=cfengine-hub registry.access.redhat.com/ubi9-init /usr/sbin/init
+```
+
+Prepare the container for cfengine
+
+```command
+$ docker exec cfengine-hub bash -c "dnf -y update; dnf -y install procps-ng iproute"
+```
+
+Install cfengine-community package
+
+```command
+$ docker exec cfengine-hub bash -c "dnf -y install https://cfengine-package-repos.s3.amazonaws.com/community_binaries/Community-3.24.0/agent_rhel9_x86_64/cfengine-community-3.24.0-1.el9.x86_64.rpm"
+```
+
+Bootstrap cf-agent
+
+```command
+$ docker exec cfengine-hub bash -c "/usr/local/sbin/cf-agent --bootstrap \$(ip -4 -o addr show eth0 | awk '{print \$4}' | cut -d'/' -f1)"
+```
+
+## Preparing CFEngine host in container
+The procedure is similar to the cfengine-hub deployment, with changes to the name of the container for identification.
+
+```command
+$ docker run --privileged -dit --name=cfengine-host registry.access.redhat.com/ubi9-init /usr/sbin/init
+```
+
+Rest of the installation steps remain the same as for **cfengine-hub**.
+
+### Bootstrap cfengine-host to the policy server container.
+Find ip of **cfengine-hub**:
+
+```command
+$ CFENGINE_HUB_IP=$(docker exec cfengine-hub bash -c "ip -4 -o addr show eth0 | awk '{print \$4}' | cut -d'/' -f1")
+```
+
+Bootstrap cfengine-host to cfengine-hub:
+
+```command
+$ docker exec cfengine-host bash -c "/usr/local/sbin/cf-agent --bootstrap ${CFENGINE_HUB_IP}"
```

Comment:
```suggestion
docker exec cfengine-host bash -c "/usr/local/sbin/cf-agent --bootstrap ${CFENGINE_HUB_IP}"
```

Reply:
Humm, this didn't work for me.

```
docker exec cfengine-host bash -c "/usr/local/sbin/cf-agent --bootstrap ${CFENGINE_HUB_IP}"
bash: line 1: /usr/local/sbin/cf-agent: No such file or directory

```

Reply:
Yeah, that's because we miss to read
> Rest of the installation steps remain the same as for **cfengine-hub**.

I've removed this and added granular steps instead.

After the fix
```
cfengine$ docker exec cfengine-host bash -c "/usr/local/sbin/cf-agent --bootstrap ${CFENGINE_HUB_IP}"
  notice: Bootstrap mode: implicitly trusting server, use --trust-server=no if server trust is already established
  notice: Trusting new key: MD5=e5da40ee8037819f033e972d0fa24bcf
R: Bootstrapping from host '172.17.0.2' via built-in policy '/var/cfengine/inputs/failsafe.cf'
R: This autonomous node assumes the role of voluntary client
R: Updated local policy from policy server
R: Triggered an initial run of the policy
R: Restarted systemd unit cfengine3
  notice: Bootstrap to '172.17.0.2' completed successfully
```

While working on this part I realized that the Dockerfile and `docker compose` based approach is much efficient and faster as the installation happens only once.
However, for understanding sake, we should maintain this approach for deploying containers individually as well.